File: toDoApp.py
tasks = []


def add_task(task):
    tasks.append(task.strip())
    print("task added!")


def show_tasks():
    if not tasks:
        print("no tasks yet")
    else:
        for i in range(len(tasks)):
            print(i + 1, ".", tasks[i])


def remove_task(task_number):
    if 1 <= task_number <= len(tasks):
        tasks.pop(task_number - 1)
        print("task removed!!")
    else:
        print("Invalid task number.")


def main():
    while True:
        print("1. Add Task")
        print("2. Show Tasks")
        print("3. Remove Task")
        print("4. Exit")
        ch = input("enter choice : ").strip()
        if ch == "1":
            t = input("enter task : ").strip()
            add_task(t)
        elif ch == "2":
            show_tasks()
        elif ch == "3":
            if not tasks:
                print("No tasks to remove.")
                continue
            try:
                n = int(input("enter task no to remove: "))
                remove_task(n)
            except ValueError:
                print("Please enter a valid number.")
        elif ch == "4":
            break
        else:
            print("wrong choice!!")


if __name__ == "__main__":
    main()

File: ToDoApp.java
/// ToDoApp.java
import java.util.ArrayList;
import java.util.InputMismatchException;
import java.util.Scanner;

public class ToDoApp {
    static ArrayList<String> tasks = new ArrayList<>(); // used generics for better stability

    public static void addtask(String t) {
        tasks.add(t);
        System.out.println("Task Added!!");
    }

    public static void showTasks() {
        if (tasks.isEmpty()) {
            System.out.println("No tasks yet");
        } 
        else {
            for (int i = 0; i < tasks.size(); i++) {
                System.out.println(
                (i + 1) + ". " + tasks.get(i)); // fixed because i goes up to tasks.size()
            } // fixes java.lang.IndexOutOfBoundsException
        }
    }

    public static void removeTask(int n) {
        if (tasks.isEmpty()) { // checks if list is empty then prints accordingly
            System.out.println("Nothing to remove");
        } 
        else if (n > 0 && n <= tasks.size()) {
            tasks.remove(n - 1);
            System.out.println("Task removed");
        } 
        else {
            System.out.println("Invalid task number");
        }
  }


    public static void main(String args[]) {
        Scanner s = new Scanner(System.in);
        boolean exit = false;
        while (!exit) {
            System.out.println("------ To-Do App ------");
            System.out.println("1. Add Task"); 
            System.out.println("2. Show Tasks");
            System.out.println("3. Remove Task");
            System.out.println("4. Exit");

            try {
                int choice = s.nextInt();
                s.nextLine(); 
                switch(choice) {
                    case 1:
                        System.out.println("Enter Task: ");
                        String t = s.nextLine();
                        addtask(t);
                        System.out.println();
                        break;
                    case 2:
                        showTasks();
                        System.out.println();
                        break;
                    case 3:
                        System.out.print("Enter task no to remove: ");
                        int n = s.nextInt();
                        System.out.print("Are you sure you want to remove task number " + n + ". " + tasks.get(n-1) + "? (y/n): ");
                        String ch = s.next();
                        if(ch.equalsIgnoreCase("y")){
                            removeTask(n);
                        }
                        System.out.println("");
                        break;
                    case 4:
                        exit = true;
                        break;
                    default:
                        System.out.println("Wrong Choice!!");
                        System.out.println();
                        break;
                }
            } 
            catch (InputMismatchException e1) {
                System.err.println("Invalid Input!");
                s.nextLine();
                System.out.println("");
            }
        }
    }
}
